modules/xcode/xcode_project.lua

```diff
@@ -15,6 +15,39 @@
 	local fileconfig = p.fileconfig
 	local tree = p.tree
 
+--
+-- Checks if a node must be excluded completely from a target or not. It will
+-- return true only if the node has the "ExcludeFromBuild" flag in all the
+-- configurations.
+--
+-- @param node
+--    The node to check.
+-- @param prj
+--    The project being generated.
+-- @returns
+--    A boolean, telling whether the node must be excluded from its target or not.
+--
+	function xcode.mustExcludeFromTarget(node, prj)
+		if not node.configs then
+			return false
+		end
+
+		local value
+		for cfg in premake.project.eachconfig(prj) do
+			local filecfg = premake.fileconfig.getconfig(node, cfg)
+			if filecfg then
+				local newValue = not not filecfg.flags.ExcludeFromBuild
+				if value == nil then
+					value = newValue
+				elseif value ~= newValue then
+					print("WARNING: " .. node.name .. " is excluded in just some configurations. Autocompletion will not work correctly on this file in Xcode.")

```

Comment:
This should probably use `p.warn` instead?

Reply:
Ok, changed.